robot_smach_states/src/robot_smach_states/manipulation/active_grasp_detector.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import absolute_import
+
+# ROS
+import rospy
+import smach
+
+# TU/e Robotics
+from robot_skills.arm.arms import GripperTypes
+
+
+class ActiveGraspDetector(smach.State):
+    REQUIRED_ARM_PROPERTIES = {"required_gripper_types": [GripperTypes.GRASPING], }
+    
+    def __init__(self, arm_designator, robot):
+        """
+        State for detecting whether or not the robot is holding something
+        
+        :param arm_designator: designator that resolves to arm to check
+        :param robot: Robot to execute the state with
+        :param timeout: float amount of time the procedure may take
+        """
+        smach.State.__init__(self, outcome=['succeeded', 'failed', 'timeout'])
+        self.robot = robot
+        
+        self.arm_designator = arm_designator
+        
```

Comment:
Values for the detection have to be tested to see if there are better candidates